src/app/modules/verification/verification.controller.ts

```diff
@@ -168,6 +175,81 @@ export const handleGetOtp: RequestHandler<
     })
 }
 
+/**
+ * NOTE: This is exported solely for testing
+ * Generates an otp when a user requests to verify a field.
+ * The current answer is signed, and the signature is also saved in the transaction, with the field id as the key.
+ * @param answer The mobile or email number of the user
+ * @param transactionId The id of the transaction to verify
+ * @param formId The id of the form to verify
+ * @param fieldId The id of the field to verify
+ * @returns 201 when otp generated successfully
+ * @returns 400 when the parameters could not be parsed
+ * @returns 400 when the transaction has expired
+ * @returns 400 when the otp could not be sent via sms
+ * @returns 400 when the otp could not be sent via email
+ * @returns 400 when the provided phone number is not valid
+ * @returns 400 when the field type is not supported for validation
+ * @returns 404 when the requested form was not found
+ * @returns 404 when the transaction was not found
+ * @returns 404 when the field was not found
+ * @returns 422 when the user requested for a new otp without waiting
+ * @returns 500 when the otp could not be hashed
+ * @returns 500 when there is a database error
+ */
+export const _handleGenerateOtp: RequestHandler<
+  { transactionId: string; formId: string; fieldId: string },
+  ErrorDto,
+  { answer: string }
+> = async (req, res) => {
+  const { transactionId, formId, fieldId } = req.params
+  const { answer } = req.body
+  const logMeta = {
+    action: 'handleGenerateOtp',
+    transactionId,
+    fieldId,
+    ...createReqMeta(req),
+  }
+  return FormService.retrieveFormById(formId)
+    .andThen(() => generateOtpWithHash(logMeta, SALT_ROUNDS))
+    .andThen(({ otp, hashedOtp }) =>
+      VerificationFactory.sendNewOtp({
+        fieldId,
+        hashedOtp,
+        otp,
+        recipient: answer,
+        transactionId,
+      }),
+    )
+    .map(() => res.sendStatus(StatusCodes.CREATED))
+    .mapErr((error) => {
+      logger.error({
+        message: 'Error creating new OTP',
+        meta: logMeta,
+        error,
+      })
+      const { errorMessage, statusCode } = mapRouteError(error)
+      return res.status(statusCode).json({ message: errorMessage })
+    })
+}
+
+/**
+ * Handler for the POST /forms/:formId/fieldverifications/:transactionId/fields/:fieldId/otp/generate endpoint
+ */
+export const handleGenerateOtp = [
+  celebrate({
+    [Segments.BODY]: Joi.object({
+      fieldType: Joi.string().valid(...VERIFIED_FIELDTYPES),
+      answer: Joi.string().when('fieldType', {
+        is: VerifiableFieldType.Email,
+        then: Joi.string().email(),
+        otherwise: Joi.string().pattern(/\+(65)\d{8}/m),
```

Comment:
I think we do allow OTPs to international numbers right? so this might be too strict

Reply:
oh dang never knew that - it's for all international numbers right?

Reply:
I think so, @karrui can confirm?

Reply:
ya, OTPs can be any number

Reply:
kk, won't raise the issue then but will change validation to all strings